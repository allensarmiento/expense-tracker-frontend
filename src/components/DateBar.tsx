import Container from 'react-bootstrap/Container';
import Button from 'react-bootstrap/Button';
import styled from 'styled-components';
import { DateFormat } from '../services/format/DateFormat';

const DateContainer = styled(Container)`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0;
`;

const ArrowButton = styled(Button)`
  font-size: 3rem;
`;

const Heading = styled.div`
  padding: 1.2rem 1.8rem;
  margin: 0;
  background-color: #fff;
  font-size: 3rem;
`;

type DateProps = {
  year: number;
  month: number;
};

const DateBar = ({ year, month }: DateProps) => {
  const dateFormat = new DateFormat(year, month);

  return (
    <DateContainer>
      <ArrowButton variant="light">&larr;</ArrowButton>
      <Heading>{dateFormat.month} {dateFormat.year}</Heading>
      <ArrowButton variant="light">&rarr;</ArrowButton>
    </DateContainer>
  );
};

export default DateBar;
