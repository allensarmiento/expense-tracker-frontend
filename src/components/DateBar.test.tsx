import { shallow } from 'enzyme';
import DateBar from './DateBar';

it('renders without crashing', () => {
  expect(shallow(
    <DateBar year={2021} month={2} />
  )).toMatchSnapshot();
});
