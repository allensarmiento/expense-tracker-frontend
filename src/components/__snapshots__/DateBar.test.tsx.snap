// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Styled(Container)>
  <Styled(Button)
    variant="light"
  >
    ←
  </Styled(Button)>
  <styled.div>
    February
     
    2021
  </styled.div>
  <Styled(Button)
    variant="light"
  >
    →
  </Styled(Button)>
</Styled(Container)>
`;
